numpy/lib/npyio.py

```diff
@@ -89,7 +91,17 @@ def __dir__(self):
 def zipfile_factory(*args, **kwargs):
     import zipfile
     kwargs['allowZip64'] = True
-    return zipfile.ZipFile(*args, **kwargs)
+    if sys.version_info[:2]==(2,6):
+        # ZipFile class compatible with the 'with' statement
+        # In python => 2.7 ZipFile class has been corrected
+        class CompatZipFile(zipfile.ZipFile):
+            def __enter__(self):
+                return self
+            def __exit__(self, exc_type, exc_val, exc_tb):
+                self.close()
+        return CompatZipFile(*args, **kwargs)        
```

Comment:
We should use contextlib.closing decorator instead here as suggested by @wackywendell